lib/web_ui/lib/src/engine/platform_views.dart

```diff
@@ -5,90 +5,7 @@
-
-  /// Private constructor so this class can be a singleton.
-  PlatformViewRegistry._();
-
-  /// Register [viewTypeId] as being creating by the given [factory].
-  bool registerViewFactory(String viewTypeId, PlatformViewFactory factory) {
-    if (registeredFactories.containsKey(viewTypeId)) {
-      return false;
-    }
-    registeredFactories[viewTypeId] = factory;
-    return true;
-  }
-
-  /// Returns the view that has been created with the given [id], or `null` if
-  /// no such view exists.
-  html.Element getCreatedView(int id) {
-    return _createdViews[id];
-  }
-}
-
-/// A function which takes a unique [id] and creates an HTML element.
-typedef PlatformViewFactory = html.Element Function(int viewId);
-
-/// The platform view registry for this app.
-final PlatformViewRegistry platformViewRegistry = PlatformViewRegistry._();
-
-/// Handles a platform call to `flutter/platform_views`.
-///
-/// Used to create platform views.
-void handlePlatformViewCall(
-  ByteData data,
-  ui.PlatformMessageResponseCallback callback,
-) {
-  const MethodCodec codec = StandardMethodCodec();
-  final MethodCall decoded = codec.decodeMethodCall(data);
-
-  switch (decoded.method) {
-    case 'create':
-      _createPlatformView(decoded, callback);
-      return;
-    case 'dispose':
-      _disposePlatformView(decoded, callback);
-      return;
-  }
-  callback(null);
-}
-
-void _createPlatformView(
-    MethodCall methodCall, ui.PlatformMessageResponseCallback callback) {
-  final Map<dynamic, dynamic> args = methodCall.arguments;
-  final int id = args['id'];
-  final String viewType = args['viewType'];
-  const MethodCodec codec = StandardMethodCodec();
-
-  // TODO(het): Use 'direction', 'width', and 'height'.
-  if (!platformViewRegistry.registeredFactories.containsKey(viewType)) {
-    callback(codec.encodeErrorEnvelope(
-      code: 'Unregistered factory',
-      message: "No factory registered for viewtype '$viewType'",
-    ));
-    return;
-  }
-  // TODO(het): Use creation parameters.
-  final html.Element element =
-      platformViewRegistry.registeredFactories[viewType](id);
-
-  platformViewRegistry._createdViews[id] = element;
-  callback(codec.encodeSuccessEnvelope(null));
-}
-
-void _disposePlatformView(
-    MethodCall methodCall, ui.PlatformMessageResponseCallback callback) {
-  final int id = methodCall.arguments;
-  const MethodCodec codec = StandardMethodCodec();
-
-  // Remove the root element of the view from the DOM.
-  platformViewRegistry._createdViews[id]?.remove();
-  platformViewRegistry._createdViews.remove(id);
-
-  callback(codec.encodeSuccessEnvelope(null));
-}
+// TODO(yjbanov): The code in this file was temporarily moved to lib/web_ui/lib/ui.dart
+//                during the NNBD migration so that `dart:ui` does not have to export
+//                `dart:_engine`. NNBD does not allow exported non-migrated libraries
+//                from migrated libraries.
```

Comment:
@leafpetersen - is this a bug or an intended restriction?

Reply:
This was intentional.  We could remove the restriction, but I was aiming to have the property that opted in packages had no legacy types in their public interface, so that if you migrated relative to an opted in package you could be confident that you were done.